### PR TITLE
Remote repository: Add user admin action for syncing remote repositories

### PR DESCRIPTION
**Pull request by saadmk11**
This PR implements a User Admin Action to run `sync_remote_repositories()` task for selected users.

Closes https://github.com/readthedocs/readthedocs.org/issues/9269

### CODE REVIEW

File: readthedocs/core/admin.py [hunk 1]
```diff
@@ -94,6 +98,32 @@ def ban_user(self, request, queryset):
...
+                )
+            )
+
+        self.message_user(
+            request,
+            mark_safe(
+                'Following sync remote repository tasks were '
+                'triggered: {}'.format(', '.join(formatted_task_urls))
+            )
+        )
```
Comment (humitos) [inline readthedocs/core/admin.py:123]: I'm not sure these links are useful/required. @agjohnson what do you think?

File: readthedocs/core/utils/tasks/permission_checks.py [hunk 2]
```diff
@@ -11,3 +9,10 @@ def user_id_matches(request, state, context):  # pylint: disable=unused-argument
         if request.user.id == user_id:
             return True
     return False
+
+
+def user_id_matches_or_admin(request, state, context):
```
Comment (humitos) [inline readthedocs/core/utils/tasks/permission_checks.py:14]:
```suggestion
def user_id_matches_or_superuser(request, state, context):
```

File: readthedocs/rtd_tests/tests/test_user_admin_actions.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
+from unittest.mock import patch
```
Comment (humitos) [inline readthedocs/rtd_tests/tests/test_user_admin_actions.py:1]: We are creating new test files where they belong and avoiding this old pattern where all the files were all together. Can you move this file to `readthedocs/core/tests/` instead?

**Review by humitos — COMMENTED** [review]
This looks great! I'd love to see the same for an organization. That way, we can resync all the remote repositories for all the users that belong to an organization in one click. Would you like to add this?

File: readthedocs/rtd_tests/tests/test_user_admin_actions.py (hunk 3 above)
Comment (saadmk11) [reply readthedocs/rtd_tests/tests/test_user_admin_actions.py:1]: Oh, sure!

File: readthedocs/core/admin.py (hunk 1 above)
Comment (saadmk11) [reply readthedocs/core/admin.py:123]: I just added this to have an easy way to quickly check the task status as I was not sure if there are any internal admin interfaces to check this for each of the triggered tasks by this action.

**Comment by saadmk11** [comment]
> This looks great! I'd love to see the same for an organization. That way, we can resync all the remote repositories for all the users that belong to an organization in one click. Would you like to add this?

@humitos I am not familiar with the `Organization` stuff. I can put together a new PR for that after I check it out. :)

**Comment by humitos** [comment]
Sounds good to me! 🤝

**Comment by ericholscher** [comment]
Letting @humitos handle review here 👍

**Review by humitos — COMMENTED** [review]
Looks good to me 👍🏼 !

`checks` tests are failing because of pre-commit. You should execute 

```
pip install pre-commit
pre-commit run --from-ref main --to-ref HEAD
```

to make the changes required.

**Comment by saadmk11** [comment]
@humitos Done!

**Review by humitos — APPROVED** [review]
🥳

**Pull request merged**